https://bugs.gentoo.org/931341
https://gitlab.winehq.org/wine/wine/-/commit/24ca848c568d5a2786abd26f2e7314b2e3f509f3
From: Fotios Valasiadis <fvalasiad@gmail.com>
Date: Sun, 5 May 2024 03:34:57 +0300
Subject: [PATCH] ntdll/unix: Fix building on musl by explicitly including
 asm/ioctls.h.

After commit 898ab8dab19d498c17859f39a55e317ee7e367a5 wine would no longer
build on musl.

Issue is that apparently TCSETS2 isn't defined when including sys/ioctl.h.

A little digging shows that glibc goes ahead and includes asm/ioctls.h in
sys/ioctl.h, providing said macro.  Musl on the other hand doesn't and relies
on bits/ioctl.h, which lacks said macro.

Signed-off-by: Fotios Valasiadis <fvalasiad@gmail.com>
--- a/dlls/ntdll/unix/serial.c
+++ b/dlls/ntdll/unix/serial.c
@@ -33,6 +33,7 @@
 #include <stdio.h>
 #ifdef HAVE_ASM_TERMBITS_H
 # include <asm/termbits.h>
+# include <asm/ioctls.h>
 #else
 # include <termios.h>
 #endif
-- 
GitLab

